src/main/java/task/Task.java

```diff
@@ -0,0 +1,79 @@
+package task;
+
+import core.ErrorMessage;
+import dukexception.DukeException;
+
+import java.io.Serializable;
+
+/**
+ * A abstract task to be inherited by specific tasks such as
+ * to do, deadline or event.
+ * Contains the description and the status of the task.
+ */
+
+public abstract class Task implements Serializable {
+
+    private String description;
+    private String typeIcon;
+    private boolean isDone;
+
+    /**
+     * Constructor for task.
+     * @param description specifies the detail of the task.
+     */
+    public Task(String description,String typeIcon) {
+        this.description = description;
+        this.isDone = false;
+        this.typeIcon=typeIcon;
+    }
+
+    /**
+     * Gets the status of the task.
+     * @return status of the task in string.
+     */
+    public String getStatusIcon() {
+        return (isDone ? "\u2713" : "\u2718"); //return tick or X symbols
+    }
+
+    /**
+     * Abstract method to get the specific type of the task.
+     * @return the specific type of the task in string.
+     */
+    public String getTypeIcon(){
+        return this.typeIcon;
+    };
+
+    /**
+     * Marks the task as done.
+     */
+    public void setDone() throws DukeException {
+        if(isDone){
```

Comment:
Another implementation is to just set isDone as done, without caring whether it is not done previously or not